### PR TITLE
Add script for updating symbols files

### PR DESCRIPTION
**Pull request by sbc100**
Also rename existing symbols files for consistency.

### CODE REVIEW

File: tools/update_symbols.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    existing_type = symbol_map.get(sym_name)
+    if not existing_type:
+      symbol_map[sym_name] = sym_type
+      continue
+    if existing_type == 'U' and sym_type != 'U':
+      symbol_map[sym_name] = sym_type
+    elif existing_type == 'W' and sym_type not in ('W', 'U'):
+      symbol_map[sym_name] = sym_type
+    elif sym_type not in ('W', 'U'):
+      # We don't know expect to see two defined version of a given symbol
```
Comment (dschuff) [inline tools/update_symbols.py]: remove 'know' ?

File: tools/update_symbols.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    print('Generating syms for: %s' % libs[0])
+    output = shared.run_process([shared.LLVM_NM, '-g', libs[0]], stdout=shared.PIPE).stdout
+  new_symbols = filter_and_sort(output)
+
+  with open(symbol_file, 'w') as f:
+    f.write(new_symbols)
+
+
+def main():
+  parser = optparse.OptionParser(description=__doc__, usage="usage: %prog [options] [<file>...]")
```
Comment (dschuff) [inline tools/update_symbols.py]: optparse is deprecated. I think for this use cases argparse is pretty much the same.

**Comment by sbc100** [comment]
OK this should be good to go now. PTAL.

**Comment by sbc100** [comment]
> In general I like this a lot. I guess all the symbols changes all just from regenerating everything?

Yeah, and because we were not doing a good job of keeping this stuff up-to-date previously.

**Comment by sbc100** [comment]
Can someone LG this?

**Pull request merged**